### PR TITLE
test(hx-tooltip): implement component tests

### PR DESCRIPTION
**Pull request by BY00565233**
## Description

Test Case `<hx-tooltip>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-2034

## Before you request a review for this PR:

- [x] Did you add component tests for any new code?
- [x] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [x] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

**Review by lalithkarikelli — APPROVED** [review]
DEV LGTM

File: src/elements/hx-tooltip/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+            });
+        });
+    });
+
+    describe('verify onCreate method', () => {
+        it('test for default position', async () => {
+            const component = /** @type {HXTooltipElement} */ await fixture(template);
+            const defaultPosition = 'top-center';
+            const attr = component.hasAttribute('position');
+            const bottomStart = component.getAttribute('position');
```
Comment (100stacks) [inline src/elements/hx-tooltip/index.spec.js:109]: Please change variable to `tooltipPosition`.

File: src/elements/hx-tooltip/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        });
+    });
+
+    describe('test <hx-tooltip> getters and setters', () => {
+        it('should be able to set for', async () => {
+            const component = /** @type {HXTooltipElement} */ await fixture(template);
+            const attr = component.for;
+
+            expect(attr).to.not.be.null;
+        });
```
Comment (100stacks) [inline src/elements/hx-tooltip/index.spec.js:162]: Please update to test:

https://github.com/HelixDesignSystem/helix-ui/blob/6ac257bf730b79eb7b55da76800a7e39def34f85/src/elements/hx-tooltip/index.js#L88-L112

* get controlElement
* get htmlFor
* set htmlFor
* setShadowPosition

**Review by 100stacks — CHANGES_REQUESTED** [review]
### Good start! Please make the few changes below.

File: src/elements/hx-tooltip/index.spec.js (hunk 1 above)
Comment (BY00565233) [reply src/elements/hx-tooltip/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-tooltip/index.spec.js (hunk 2 above)
Comment (BY00565233) [reply src/elements/hx-tooltip/index.spec.js:162]: Changes have been done, Please review

File: src/elements/hx-tooltip/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        it('should get control element id of iconCtrlEl', async () => {
+            const fragment = /** @type {HXTooltipElement} */ await fixture(mockupId);
+            const id = 'iconCtrlEl';
+            const ctrlElement = fragment.controlElement;
+            const ctrlElementID = ctrlElement.id;
+
+            expect(ctrlElementID).to.equal(id);
+        });
+    });
+
```
Comment (100stacks) [inline src/elements/hx-tooltip/index.spec.js:196]: @BY00565233, please see updates to the test suite.

**Review by 100stacks — CHANGES_REQUESTED** [review]
### @BY00565233, see comments, squash, and update screenshot.

File: src/elements/hx-tooltip/index.spec.js (hunk 3 above)
Comment (BY00565233) [reply src/elements/hx-tooltip/index.spec.js]: Done

File: src/elements/hx-tooltip/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+            Top Center Tooltip
+        </hx-tooltip>`;
+    const mockupId = `
+        <p>
+            <hx-icon
+                id="icon1"
+                type="help-circle"
+            >
+            </hx-icon>
+        </p>`;
```
Comment (100stacks) [inline src/elements/hx-tooltip/index.spec.js:28]: Please change:

```js
const mockupId = `
    <hx-tooltip for="iconCtrlEl" position="bottom-center">
        Bottom Center Tooltip
    </hx-tooltip>
    <p>
        <hx-icon id="iconCtrlEl" type="help-circle"></hx-icon>
    </p>`;
```

File: src/elements/hx-tooltip/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+            fragment.setShadowPosition(newPosition);
+
+            const tooltipPosition = fragment.shadowRoot.querySelector('div').getAttribute('position');
+
+            expect(tooltipPosition).to.equal(newPosition);
+        });
+
+        it('should get control element id of icon1', async () => {
+            const fragment = /** @type {HXTooltipElement} */ await fixture(mockupId);
+            const id = 'icon1';
```
Comment (100stacks) [inline src/elements/hx-tooltip/index.spec.js:190]: Please change based on the updates:

```js
const id = 'iconCtrlEl';
```

File: src/elements/hx-tooltip/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+            const ctrlElement = fragment.querySelector('hx-icon');
```
Comment (100stacks) [inline src/elements/hx-tooltip/index.spec.js:191]: Please change based on the updates:

```js
const ctrlElement = fragment.controlElement;
```

File: src/elements/hx-tooltip/index.spec.js [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+            const newPosition = 'bottom-center';
+
+            fragment.setShadowPosition(newPosition);
+
+            const tooltipPosition = fragment.shadowRoot.querySelector('div').getAttribute('position');
+
+            expect(tooltipPosition).to.equal(newPosition);
+        });
+
+        it('should get control element id of icon1', async () => {
```
Comment (100stacks) [inline src/elements/hx-tooltip/index.spec.js:188]: Please change test description based on the updates below.

**Review by 100stacks — CHANGES_REQUESTED** [review]

File: src/elements/hx-tooltip/index.spec.js (hunk 7 above)
Comment (BY00565233) [reply src/elements/hx-tooltip/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-tooltip/index.spec.js (hunk 6 above)
Comment (BY00565233) [reply src/elements/hx-tooltip/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-tooltip/index.spec.js (hunk 5 above)
Comment (BY00565233) [reply src/elements/hx-tooltip/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-tooltip/index.spec.js (hunk 4 above)
Comment (BY00565233) [reply src/elements/hx-tooltip/index.spec.js]: Changes have been done, Please review

**Pull request merged**